wordpress-mobile/WordPress-iOS
# Fix date medium string warnings

This PR fixes all of the medium string warnings we currently have in the project.

**To test**

- Check over the code changes. I removed one method entirely because it didn't appear to be used any more.
- Build and run, and if you like verify dates still appear correctly in each of the following situations:
- In the Posts list, on post cards
- In the Media section, select a media item and check the uploaded date
- Post revision rows and detail view
- Reader detail date labels and the accessibility label read when selecting the header in Voiceover
- Reader post cards
- I tried to verify the changes to notifications sent by the Share Extension, but when my posts updated they always just said 'now'. The code looks fine to me though.

## Regression Notes
1. Potential unintended areas of impact

Only areas where these calls are used – however, I think they're all used for non-critical visual elements, so I think it should be fine.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I verified each of the areas changed.

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:
